staker/stakerapp.go

```diff
@@ -506,6 +507,62 @@ func (app *App) mustSetTxSpentOnBtc(hash *chainhash.Hash) {
 	}
 }
 
+// SendPhase1Transaction receives the BTC staking transaction hash that
+// should be already in BTC and creates the necessary data to submit
+// the BTC delegation into the consumer chain
+func (app *App) SendPhase1Transaction(
+	stakerAddr btcutil.Address,
+	stkTxHash *chainhash.Hash,
+	tag []byte,
+	covenantPks []*secp256k1.PublicKey,
+	covenantQuorum uint32,
+) error {
+	stkTx, err := app.wc.Tx(stkTxHash)
+	if err != nil {
+		app.logger.WithError(err).Info("err get tx details")
+		return err
+	}
+
+	wireStkTx := stkTx.MsgTx()
+	parsedStakingTx, err := staking.ParseV0StakingTx(wireStkTx, tag, covenantPks, covenantQuorum, app.network)
+	if err != nil {
+		app.logger.WithError(err).Info("err parse staking Tx with global params")
+		return err
+	}
+
+	// unlock wallet to generate pop
+	// TODO: consider unlock/lock with defer
+	err = app.wc.UnlockWallet(defaultWalletUnlockTimeout)
+	if err != nil {
+		return err
+	}
+
+	pop, err := app.createPop(stakerAddr)
+	if err != nil {
+		return err
+	}
+
+	if err := app.txTracker.AddTransactionSentToBTC(
+		wireStkTx,
+		uint32(parsedStakingTx.StakingOutputIdx),
+		parsedStakingTx.OpReturnData.StakingTime,
+		[]*btcec.PublicKey{parsedStakingTx.OpReturnData.FinalityProviderPublicKey.PubKey},
+		babylonPopToDBPop(pop),
+		stakerAddr,
+	); err != nil {
+		app.logger.WithError(err).Info("err to set tx as sent on BTC")
+		return err
+	}
+
+	stakingParams, err := app.babylonClient.Params()
+	if err != nil {
+		return err
+	}
+
+	// pushes the confirmation into the btc-staker mechanism and verifies if it is deep enough
+	return app.waitForStakingTransactionConfirmation(stkTxHash, parsedStakingTx.StakingOutput.PkScript, stakingParams.ConfirmationTimeBlocks, app.currentBestBlockHeight.Load())
```

Comment:
hmm the main disadvantage to this approach is that we return to the caller before even sending to Babylon, so if there are some problems along the way (like wallet not able to sign some transaction etc.) then we end up with invalid delegation stored in db.

Do avoid this we could create new command handled in `handleStakingCommands` like `migrateStakingCmd` and return to the caller only after we send to Babylon (handling of this command would be super similiar to `handlePreApprovalCmd`  ).  Wdyt ?

Reply:
The current command of `stakingRequestedCmdChan` also returns before send to Babylon, we only wait to send to BTC the rest of the process is done internally [stakingTxHash](https://github.com/babylonlabs-io/btc-staker/blob/15cb74d032a5d123e38ac930cc20db98e2528272/staker/stakerapp.go#L1498) is the BTC tx hash

Reply:
yup this is partially true:
1. for pre-approval flow we return to the caller only after send to Babylon is success
2. for post-approval flow we return after send to BTC, but for post approval process we build whole transactions ourselves from scratch so we knot wallet is maintianing all staker key.

So alternative approach (instead of implementing new command) would be to leave it as it is but add additional check in `SendPhase1Transaction` that:
1. retrieves public key corresponding to provided `stakerAddr` - there is `AddressPublicKey` function
2. check whether public key retrieved is equal to staker public key which is returned from `ParseV0StakingTx` 
This way we can be sure that connected bitcoind wallet maintains correct key, that must make signatures on slashing and unbonding transaction

Reply:
You are correct, the PreApproval does send and wait for Babylon, but we do not send the tx of Babylon to the user, I will make a new command that returns the Babylon tx hash